app/engine/RowingStatistics.js

```diff
@@ -140,88 +251,108 @@ function createRowingStatistics (config) {
+      return true
+    } else {
+      return false
     }
   }
 
-  function startTraining () {
-    sessionState = 'rowing'
+  function measureRecoveryHR () {
+    // This function is called when the rowing session is stopped. postExerciseHR[0] is the last measured excercise HR
+    // Thus postExerciseHR[1] is Recovery HR after 1 min, etc..
+    if (heartrate !== undefined && heartrate > config.userSettings.restingHR && sessionStatus !== 'Rowing') {
+      log.debug(`*** HRR-${postExerciseHR.length}: ${heartrate}`)
+      postExerciseHR.push(heartrate)
+      if ((postExerciseHR.length > 1) && (postExerciseHR.length <= 4)) {
+        // We skip reporting postExerciseHR[0] and only report measuring postExerciseHR[1], postExerciseHR[2], postExerciseHR[3]
+        emitter.emit('HRRecoveryUpdate', postExerciseHR)
+      }
+      if (postExerciseHR.length < 4) {
+        // We haven't got three post-exercise HR measurements yet, let's schedule the next measurement
+        setTimeout(measureRecoveryHR, 60000)
+      }
+    }
   }
 
-  function stopTraining () {
-    sessionState = 'stopped'
-    if (rowingPausedTimer)clearInterval(rowingPausedTimer)
+  function emitWebMetrics () {
+    emitMetrics('webMetricsUpdate')
   }
 
-  function resetTraining () {
-    stopTraining()
-    distanceTotal = 0.0
-    strokesTotal = 0
-    caloriesTotal = 0.0
-    durationTotal = 0
-    caloriesAveragerMinute.reset()
-    caloriesAveragerHour.reset()
-    strokeAverager.reset()
-    powerAverager.reset()
-    speedAverager.reset()
-    powerRatioAverager.reset()
-    sessionState = 'waitingForStart'
+  function emitPeripheralMetrics () {
+    emitMetrics('peripheralMetricsUpdate')
   }
 
-  // clear the metrics in case the user pauses rowing
-  function pauseRowing () {
-    strokeAverager.reset()
-    powerAverager.reset()
-    speedAverager.reset()
-    powerRatioAverager.reset()
-    lastStrokeState = 'RECOVERY'
-    sessionState = 'paused'
-    emitter.emit('rowingPaused')
+  function emitMetrics (emitType = 'webMetricsUpdate') {
+    emitter.emit(emitType, getMetrics())
   }
 
-  // converts a timestamp in seconds to a human readable hh:mm:ss format
+  function getMetrics () {
+    const cyclePace = cycleLinearVelocity.clean() !== 0 && cycleLinearVelocity.raw() > 0 && sessionStatus === 'Rowing' ? (500.0 / cycleLinearVelocity.clean()) : Infinity
+    return {
+      sessiontype: session.targetDistance > 0 ? 'Distance' : (session.targetTime > 0 ? 'Time' : 'JustRow'),
+      sessionStatus,
+      strokeState: rower.strokeState(),
+      totalMovingTime: totalMovingTime > 0 ? totalMovingTime : 0,
+      totalMovingTimeFormatted: session.targetTime > 0 ? secondsToTimeString(Math.round(Math.max(session.targetTime - totalMovingTime), 0)) : secondsToTimeString(Math.round(totalMovingTime)),
```

Comment:
`secondsToTimeString` does not take two arguments, so the trailing 0 should be removed

Reply:
Nope, it should read secondsToTimeString(Math.round(Math.max(session.targetTime - totalMovingTime, 0))) (time should never dip below 0). But thanks for spotting this one, it is now fixed.